2-onchain-interactions.md

```diff
@@ -0,0 +1,284 @@
+- **Concluded**: emitted when outcomes are finalized
+
+### AssetTransferred
+
+The `AssetTransferred` event is defined as:
+
+```typescript
+event AssetTransferred(
+  // channelId funds were disbursed from
+  bytes32 indexed channelId,
+  // external address funds were sent to
+  bytes32 indexed destination,
+  // amount disbursed
+  uint256 amount
+);
+```
+
+This event is emitted at the completion of the `transferAll` or `claimAll` functions in the `AssetHolder.sol`. The `transferAll` method is used to disburse funds from ledger or direct channels, while the `claimAll` method will disburse funds to the appropriate guarantor channel. In the process of removing funds from a channel, the state must be finalized between participants onchain using `pushOutcome` (finalization via adjudication) or `conclude` (happy case finalization).
+
+In practice, there are helper methods `pushOutcomeAndTransferAll` as well as `concludePushOutcomeAndTransferAll` that set the onchain state as well as transfer in one onchain transaction.
+
+### Deposited
+
+The `Deposited` event is defined as:
+
+```typescript
+event Deposited(
+  // channelId to be credited
+  bytes32 indexed destination,
+  // the amount that was credited to the channel
+  uint256 amountDeposited,
+  // the channel balance post-deposit
+  uint256 destinationHoldings
+);
+```
+
+This event is emitted at the completion of the `deposit` function in both the `ERC20AssetHolder.sol` and the `ETHAssetHolder.sol` (which inherit the `AssetHolder.sol` contract). In the process of funding a channel, the funder must show their counterparty their intent to fund the channel. Both wallets will then expect a corresponding `Deposited` event to be emitted by the contracts before continuing with channel updates.
+
+### ChallengeRegistered
+
+The `ChallengeRegistered` event is defined as:
+
+```typescript
+event ChallengeRegistered(
+  // channelId to be adjudicated
+  bytes32 indexed channelId,
+  // latest nonce
+  uint48 turnNumRecord,
+  // unix timestamp when challenge will finalize
+  uint48 finalizesAt,
+  // address of participant who registered the challenge
+  address challenger,
+  // whether challenge state is final
+  bool isFinal,
+  // constant channel properties
+  FixedPart fixedPart,
+  // variable channel properties
+  ForceMoveApp.VariablePart[] variableParts,
+  // signatures of channel participants
+  Signature[] sigs,
+  // information to relate signatures to participants
+  uint8[] whoSignedWhat
+);
+```
+
+This event is emitted at the end of the `foceMove` function in `ForceMove.sol`. It provides all of the information needed for a channel participant to construct a new state when responding to a challenge. This event is emitted both on challenge creation, and challenge response if the participant is trying to play out the channel states onchain rather than resume offchain operations.
```

Comment:
```suggestion
This event is emitted at the end of the `forceMove` function in `ForceMove.sol`. It provides all of the information needed for a channel participant to construct a new state when responding to a challenge. This event is emitted both on challenge creation, and challenge response if the participant is trying to play out the channel states onchain rather than resume offchain operations.
```